Sources/EduIDExpansion/Controllers/Coordinators/MainCoordinator.swift

```diff
@@ -1,17 +1,70 @@
+    init(homeNavigationController: UINavigationController) {
+        self.homeNavigationController = homeNavigationController
+        
     }
     
     func start() {
-        let onboardingCoordinator = OnboardingCoordinator(navigationController: self.navigationController)
+        let onboardingNavigationController = UINavigationController()
+        onboardingNavigationController.modalTransitionStyle = .flipHorizontal
+        onboardingNavigationController.isModalInPresentation = true
+        
+        let onboardingCoordinator = OnboardingCoordinator(navigationController: onboardingNavigationController)
+        onboardingCoordinator.parent = self
         children.append(onboardingCoordinator)
+        
+        homeNavigationController.present(onboardingNavigationController, animated: false)
         onboardingCoordinator.start()
     }
+    
+    func showPersonalInfo() {
+        let editPersonalInfoViewcontroller = EditPersonalInfoViewController()
+        editPersonalInfoViewcontroller.screenType = .personalInfoScreen
+        editPersonalInfoViewcontroller.modalTransitionStyle = .flipHorizontal
+        
+        let personalInfoNavigationController = UINavigationController(rootViewController: editPersonalInfoViewcontroller)
+        personalInfoNavigationController.modalTransitionStyle = .flipHorizontal
+        personalInfoNavigationController.isModalInPresentation = true
+        
+        let personalInfoCoordinator = PersonalInfoCoordinator(navigationController: personalInfoNavigationController)
+        editPersonalInfoViewcontroller.coordinator = personalInfoCoordinator
+        children.append(personalInfoCoordinator)
+         
+        homeNavigationController.present(personalInfoNavigationController, animated: true)
+        personalInfoCoordinator.start()
+    }
+    
+    //MARK: - start scan screen
+    @objc
+    func showScanScreen() {
+        let scanViewcontroller = ScanViewController(viewModel: ScanViewModel())
+        scanViewcontroller.coordinator = self
+        let logo = UIImageView(image: UIImage.eduIDLogo)
+        logo.width(92)
+        logo.height(36)
+        scanViewcontroller.navigationItem.titleView = logo
+        scanViewcontroller.navigationItem.hidesBackButton = true
+        scanViewcontroller.navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage.arrowBack, style: .plain, target: self, action: #selector(dismissScanScreen))
+        
+        scanViewcontroller.navigationItem.leftBarButtonItem?.image = scanViewcontroller.navigationItem.leftBarButtonItem?.image?.withRenderingMode(.alwaysTemplate)
+        
+        let scanNavigationController = UINavigationController(rootViewController: scanViewcontroller)
+        scanNavigationController.navigationBar.tintColor = .white
+        scanNavigationController.setNavigationBarHidden(false, animated: true)
+        homeNavigationController.present(scanNavigationController, animated: true)
```

Comment:
I'd suggest adding a ScanCoordinator, as you may need to start that flow from different points in the app.

Reply:
I implemented that